controllers/machineset_controller.go

```diff
@@ -549,3 +542,118 @@ func (r *MachineSetReconciler) hasMatchingLabels(machineSet *clusterv1.MachineSe
 func (r *MachineSetReconciler) shouldAdopt(ms *clusterv1.MachineSet) bool {
 	return !util.HasOwner(ms.OwnerReferences, clusterv1.GroupVersion.String(), []string{"MachineDeployment", "Cluster"})
 }
+
+func (r *MachineSetReconciler) calculateStatus(ms *clusterv1.MachineSet, filteredMachines []*clusterv1.Machine) clusterv1.MachineSetStatus {
+	logger := r.Log.WithValues("machineset", ms.Name, "namespace", ms.Namespace)
+	newStatus := ms.Status
+
+	// Copy label selector to its status counterpart in string format.
+	// This is necessary for CRDs including scale subresources.
+	selector, _ := metav1.LabelSelectorAsSelector(&ms.Spec.Selector)
+	newStatus.Selector = selector.String()
+
+	// Count the number of machines that have labels matching the labels of the machine
+	// template of the replica set, the matching machines may have more
+	// labels than are in the template. Because the label of machineTemplateSpec is
+	// a superset of the selector of the replica set, so the possible
+	// matching machines must be part of the filteredMachines.
+	fullyLabeledReplicasCount := 0
+	readyReplicasCount := 0
+	availableReplicasCount := 0
+	templateLabel := labels.Set(ms.Spec.Template.Labels).AsSelectorPreValidated()
+
+	// Retrieve Cluster, if any.
```

Comment:
Remove the "if any" part? Or handle the error?

Reply:
This was copied from the other file, I have no idea why it ignores the error though

Reply:
We used to have a fallback to using the in-cluster client when the Cluster wasn't defined: https://github.com/kubernetes-sigs/cluster-api/blob/release-0.2/controllers/machineset_status.go#L141-L146 

I suspect this call was never updated when that was removed.

Reply:
I’ll remove it in a bit

Reply:
Done, removed the line, and passed in the cluster from the caller